packages/date-picker/test/form-input.test.js

```diff
@@ -162,51 +168,176 @@ describe('form input', () => {
+      datePicker.value = '2000-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+  });
+
+  describe('min', () => {
+    beforeEach(() => {
+      datePicker = fixtureSync(`<vaadin-date-picker min="2010-01-01"></vaadin-date-picker>`);
+    });
+
+    it('should pass validation without value', () => {
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should fail validation with a value < min', () => {
+      datePicker.value = '2000-01-01';
+      expect(datePicker.checkValidity()).to.be.false;
+      expect(datePicker.validate()).to.be.false;
+    });
+
+    it('should pass validation with a value > min', () => {
+      datePicker.value = '2020-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should pass validation with a value = min', () => {
+      datePicker.value = '2010-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should fail validation when the entered value < min', async () => {
+      setInputValue(datePicker, '1/1/2000');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.false;
+      expect(datePicker.validate()).to.be.false;
+    });
+
+    it('should pass validation when the entered value > min', async () => {
+      setInputValue(datePicker, '1/1/2022');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should pass validation when the entered value = min', async () => {
+      setInputValue(datePicker, '1/1/2022');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+  });
+
+  describe('max', () => {
+    beforeEach(() => {
+      datePicker = fixtureSync(`<vaadin-date-picker max="2010-01-01"></vaadin-date-picker>`);
+    });
+
+    it('should pass validation without value', () => {
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should fail validation with a value > max', () => {
+      datePicker.value = '2020-01-01';
+      expect(datePicker.checkValidity()).to.be.false;
+      expect(datePicker.validate()).to.be.false;
+    });
+
+    it('should pass validation with a value < max', () => {
+      datePicker.value = '2000-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should pass validation with a value = max', () => {
+      datePicker.value = '2010-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
     });
 
-    it('should not be invalid without user interactions', () => {
```

Comment:
**note:** See [the `required` section](https://github.com/vaadin/web-components/pull/4163/files#diff-746ffa5fcc10c6bab313c794447622d931d27c361bb3c354f0135efabf98046fR209) instead.